### ISSUE TITLE
implement functionality for --delta

### ISSUE THREAD

**Issue by soonhokong** [dreal/dreal3#247]
feature requested by @scungao

**Comment by soonhokong** [comment]
```
	1193 - regression_nra_delta_change_precision.smt2 (Failed)
	1199 - regression_nra_delta_ea_02.smt2 (Failed)
	1261 - regression_nra_delta_osmosis-midair.smt2 (Timeout)
	1270 - regression_nra_delta_scungao_02.smt2 (Timeout)
	1273 - regression_nra_delta_simdreal_4.smt2 (Timeout)
	1274 - regression_nra_delta_simdreal_5.smt2 (Timeout)
```

```
	2029 - regression_nra_ode_delta_pzuliani_01.smt2 (Timeout)
	2030 - regression_nra_ode_delta_pzuliani_02.smt2 (Timeout)
	2031 - regression_nra_ode_delta_quad1_1_0.smt2 (Failed)
	2032 - regression_nra_ode_delta_quad2_2_0.smt2 (Failed)
	2033 - regression_nra_ode_delta_scaled_prostate_infix_0_0.smt2 (Failed)
	2034 - regression_nra_ode_delta_scaled_prostate_infix_2_0.smt2 (Failed)
	2037 - regression_nra_ode_delta_thermostat-double-sat_1.smt2 (Failed)
	2038 - regression_nra_ode_delta_thermostat-double-sat_2.smt2 (Failed)
	2039 - regression_nra_ode_delta_thermostat-double-sat_3.smt2 (Failed)
	2040 - regression_nra_ode_delta_thermostat-double-sat_4.smt2 (Failed)
	2041 - regression_nra_ode_delta_thermostat-double-sat_5.smt2 (Failed)
	2045 - regression_nra_ode_delta_thermostat-double_4.smt2 (Failed)
	2046 - regression_nra_ode_delta_thermostat-double_5.smt2 (Failed)
	2047 - regression_nra_ode_delta_thermostat-triple-sat_1.smt2 (Failed)
	2048 - regression_nra_ode_delta_thermostat-triple-sat_2.smt2 (Failed)
	2049 - regression_nra_ode_delta_thermostat-triple-sat_3.smt2 (Failed)
	2050 - regression_nra_ode_delta_thermostat-triple-sat_4.smt2 (Timeout)
	2051 - regression_nra_ode_delta_thermostat-triple-sat_5.smt2 (Timeout)
	2055 - regression_nra_ode_delta_thermostat-triple_4.smt2 (Timeout)
	2056 - regression_nra_ode_delta_thermostat-triple_5.smt2 (Timeout)
	2057 - regression_nra_ode_delta_water-double-sat_1.smt2 (Failed)
	2058 - regression_nra_ode_delta_water-double-sat_2.smt2 (Failed)
	2059 - regression_nra_ode_delta_water-double-sat_3.smt2 (Failed)
	2060 - regression_nra_ode_delta_water-double-sat_4.smt2 (Failed)
	2061 - regression_nra_ode_delta_water-double-sat_5.smt2 (Failed)
	2065 - regression_nra_ode_delta_water-double_4.smt2 (Failed)
	2066 - regression_nra_ode_delta_water-double_5.smt2 (Failed)
	2067 - regression_nra_ode_delta_water-triple-sat_1.smt2 (Failed)
	2068 - regression_nra_ode_delta_water-triple-sat_2.smt2 (Failed)
	2069 - regression_nra_ode_delta_water-triple-sat_3.smt2 (Failed)
	2070 - regression_nra_ode_delta_water-triple-sat_4.smt2 (Failed)
	2071 - regression_nra_ode_delta_water-triple-sat_5.smt2 (Failed)
	2073 - regression_nra_ode_delta_water-triple_2.smt2 (Failed)
	2074 - regression_nra_ode_delta_water-triple_3.smt2 (Failed)
	2075 - regression_nra_ode_delta_water-triple_4.smt2 (Failed)
	2076 - regression_nra_ode_delta_water-triple_5.smt2 (Failed)
```

**Comment by soonhokong** [comment]
Closed by 8abeb182d66bfdb3519aa94525daf7706c676518, e0c334ffbf39d0dba08dd97d6d5186327350bf06, 7804257c2e53e26c929fba7bf24ffcf8f3c5bafc.

Note that it doesn't support exist-forall cases yet.

**Comment by soonhokong** [comment]
`--range-delta` works as well.
